drivers/can/can_nrf.c

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_can
+
+#include <stdint.h>
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/can.h>
+#include <zephyr/drivers/can/can_mcan.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
+
+/* nRF CAN wrapper offsets */
+#define CAN_TASKS_START	  offsetof(NRF_CAN_Type, TASKS_START)
+#define CAN_EVENTS_CORE_0 offsetof(NRF_CAN_Type, EVENTS_CORE[0])
+#define CAN_EVENTS_CORE_1 offsetof(NRF_CAN_Type, EVENTS_CORE[1])
+#define CAN_INTEN	  offsetof(NRF_CAN_Type, INTEN)
+
+struct can_nrf_config {
+	uint32_t wrapper;
```

Comment:
Just out of curiosity, what is this wrapper? Is it something related to multicore and irq?

Reply:
nope, is just the nordic-specific thin layer on top of the Bosch MCAN IP (eg IRQ enablement, events, etc.)